modules/local/bwamem2_mem.nf

```diff
@@ -1,29 +1,38 @@
-params.bwa_options = "-K 100000000 -M"
-params.sequencer = "ILLUMINA"
-
 process BWAMEM2_MEM {
-    label 'CPUS_MAX'
-
-    tag "${sample}_${run}"
+    tag "${meta.id}"
+    label 'process_high'
 
-    publishDir "${params.outdir}/bwamem2_mem", mode: 'copy'
+    publishDir "${params.outdir}/bwamem2/${meta.sample}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename ->
+                    if (filename.endsWith('.version.txt')) null
+                    else filename }
 
     input:
-        tuple val(patient), val(sample), val(run), path(read1), path(read2)
+        tuple val(meta), path(reads)
         path bwa
         path fasta
         path fai
+        val options
 
     output:
-        tuple val(patient), val(sample), val(run), path("*.bam"), path("*.bai")
+        tuple val(meta), path("*.bam"), path("*.bai")
 
     script:
     CN = params.sequencing_center ? "CN:${params.sequencing_center}\\t" : ""
```

Comment:
```suggestion
```

Reply:
This parameter should come in via arguments created from the pipeline and not hardcoded here. Not everyone will use this!